scripts/lib/CIME/XML/env_batch.py

```diff
@@ -690,9 +690,12 @@ def _submit_single_job(self, case, job, dep_jobs=None, allow_fail=False,
             sequence = (batchsubmit, submitargs, batchredirect, get_batch_script_for_job(job), run_args)
 
         submitcmd = " ".join(s.strip() for s in sequence if s is not None)
+        if submitcmd.startswith("ssh"):
+            submitcmd = submitcmd.replace("cd $CASEROOT","\'cd $CASEROOT") + "\'"
```

Comment:
Are the escape characters necessary here? Python has good support for nesting quotes:
```
s = " 'hi' "
```

Reply:
Also, I'm a bit confused that the second single quote happens outside of the replace. I think this would be more clear:
```
if submitcmd.startswith("ssh"):
    stmt_to_quote = "cd $CASEROOT"
    submitcmd = submitcmd.replace(stmt_to_quote, "'{}'".format(stmt_to_quote))
```

Reply:
I'm looking into removing the escape chars.

The second quote is outside of the replace because there is more than just cd $CASEROOT that needs to be in quotes the rest of the submitcmd needs to be in it.

Reply:
@jgfouca The escapes are required.